### ISSUE TITLE
Pickling Error with 1.8

### ISSUE THREAD

**Issue by blink1073** [sphinx-doc/sphinx#5422]
Subject:  Sphinx 1.8 introduced a build error for JupyterLab

### Problem
- It appears that in 1.8 the `url_resolver` config is trying to get pickled.  We had a lambda function, but that broke with (cf https://travis-ci.org/jupyterlab/jupyterlab/jobs/427847089#L672):

```
AttributeError: Can't pickle local object 'setup.<locals>.<lambda>'
```

Changing it to a separate function yields (cf https://travis-ci.org/jupyterlab/jupyterlab/jobs/428072324#L763 and https://github.com/jupyterlab/jupyterlab/pull/5308/files):

```
_pickle.PicklingError: Can't pickle <function url_resolver at 0x2b0913492e18>: attribute lookup url_resolver on __main__ failed
```

#### Procedure to reproduce the problem

```
git clone https://github.com/jupyterlab/jupyterlab
cd jupyterlab/docs
pip install sphinx=1.8 sphinx_rtd_theme recommonmark
make html
```

### Environment info
- OS: MacOS 10.13.6
- Python version: 3.6.3
- Sphinx version: 1.8.0

**Comment by tk0miya** [comment]
It seems the error comes from the default setting for recommonmark.AutoStructify.
I'll take a look this later.

Note: I confirmed this with following dockerfile:
```
FROM tk0miya/sphinx-html

RUN git clone https://github.com/jupyterlab/jupyterlab
WORKDIR /docs/jupyterlab/docs
RUN pip3 install sphinx==1.8 sphinx_rtd_theme recommonmark
RUN make html SPHINXBUILD='python3 -msphinx'
```

**Comment by blink1073** [comment]
Cheers!

**Comment by tk0miya** [comment]
Fixed by #5429 .
Thank you for reporting.

**Comment by blink1073** [comment]
:tada:
